src/dotnet/commands/dotnet-run3/Run3Command.cs

```diff
@@ -87,30 +71,30 @@ private ITaskItem GetRunInformation()
 
             ProjectInstance projectInstance = new ProjectInstance(Project, globalProperties, null);
 
-            BuildRequestData buildRequestData = new BuildRequestData(projectInstance, new string[] { GetRunInformationTaskName });
-            BuildParameters buildParameters = new BuildParameters();
-
-            BuildResult result = BuildManager.DefaultBuildManager.Build(buildParameters, buildRequestData);
-
-            TargetResult runInfoResult;
-            if (!result.ResultsByTarget.TryGetValue(GetRunInformationTaskName, out runInfoResult))
+            string runProgram = projectInstance.GetPropertyValue("RunCommand");
+            if (string.IsNullOrEmpty(runProgram))
             {
-                throw new InvalidOperationException($"Could not find a target named '{GetRunInformationTaskName}' in your project. Please ensure 'dotnet run' supports this project.");
-            }
+                string outputType = projectInstance.GetPropertyValue("OutputType");
 
-            if (runInfoResult.ResultCode != TargetResultCode.Success)
-            {
-                throw new InvalidOperationException($"Could not get the run information for project {Project}. An internal MSBuild error has occured" + Environment.NewLine +
-                    runInfoResult.Exception?.ToString());
+                throw new GracefulException(string.Join(Environment.NewLine,
+                    "Unable to run your project.",
+                    "Please ensure you have a runnable project type and ensure 'dotnet run' supports this project.",
+                    $"The current OutputType is '{outputType}'."));
             }
 
-            ITaskItem runInfoItem = runInfoResult.Items.FirstOrDefault(i => i.ItemSpec == projectInstance.FullPath);
-            if (runInfoItem == null)
+            string runArguments = projectInstance.GetPropertyValue("RunArguments");
+            string runWorkingDirectory = projectInstance.GetPropertyValue("RunWorkingDirectory");
+
+            string fullArguments = runArguments;
+            if (_args.Any())
             {
-                throw new InvalidOperationException($"'{GetRunInformationTaskName}' did not return an ITaskItem with the project's FullPath as the ItemSpec.");
+                fullArguments += " " + ArgumentEscaper.EscapeAndConcatenateArgArrayForProcessStart(_args);
             }
 
-            return runInfoItem;
+            CommandSpec commandSpec = new CommandSpec(runProgram, fullArguments, CommandResolutionStrategy.None);
```

Comment:
I don't think there is a "command resolution strategy" for this scenario at all.  The idea being that the MSBuild "Run" properties should tell the caller (dotnet run here) exactly what they should do to run the project.  There should be no need to resolve anything, that way the caller can be "dumb".  Also, that way no matter who the caller is (VS, dotnet run, VS Code, Project Rider, etc.) the project is run the exact same way.